internal/bminventory/inventory.go

```diff
@@ -840,13 +841,24 @@ func (b *bareMetalInventory) RegisterAddHostsClusterInternal(ctx context.Context
 func (b *bareMetalInventory) createAndUploadNodeIgnition(ctx context.Context, cluster *common.Cluster, host *models.Host) error {
 	log := logutil.FromContext(ctx, b.log)
 	log.Infof("Starting createAndUploadNodeIgnition for cluster %s, host %s", cluster.ID, host.ID)
-	address := cluster.APIVip
-	if address == "" {
-		address = swag.StringValue(cluster.APIVipDNSName)
-	}
-	ignitionBytes, err := b.IgnitionBuilder.FormatSecondDayWorkerIgnitionFile(address, host.MachineConfigPoolName)
-	if err != nil {
-		return errors.Errorf("Failed to create ignition string for cluster %s", cluster.ID)
+
+	// Specify ignition endpoint based on cluster configuration:
+	var ignitionBytes []byte
+	var err error
+	if cluster.IgnitionEndpointURL != nil {
```

Comment:
just verifying - in this manner the user will also be able to provide machine-config pool as part of the url, right?

Reply:
That's what I expected, but it seems that hypershift doesn't do that, rather they just give us a URL like https://<endpoint>/ignition
So this would work for hypershift, but maybe we need a mapping of MCP to URL?

Reply:
i think that the pool is just part of the url (@karmab @yuvalk please keep me honest here)

Reply:
In this manner the user will set just the ignition endpoint.
the user can set the machine-config pool on the host.
The full URL is the ignition-endpoint + / + machine config pool
e.g.
```
ignition endpoint: https://my-cluster.redhat.com/ignition
machine-config-pool: worker
```
Full ignition URL: `https://my-cluster.redhat.com/ignition/worker`

@machacekondra this means we need to add the host machine config pool to the ignition endpoint.

Reply:
isnt the url the other way around?
https://my-cluster.redhat.com/<pool>/<ign>
https://my-cluster.redhat.com/worker/ignition

also, I'm out of context, but what if I want to serve my ignition from somewhere else entirely? is that possible?

Reply:
By default the ignition full URL is `api-int.<cluster-domain>:22623/config/<role>`
 Where role matches the machineConfigPool

In HyperShift (the first user of this code) the  ignition endpoint is somewhere else entirely and this is what this PR aims to enable.

Side note - Specifically in HyperShift it seems that the ignition endpoint doesn't expect to get the node role in the URL (doesn't matter that much since all nodes are workers anyway)